octobot_backtesting/strategy_optimizer/strategy_optimizer.py

```diff
@@ -38,18 +38,19 @@ class StrategyOptimizer:
     """
     StrategyOptimizer is a tool that performs backtesting with different configurations
     """
-    def __init__(self, config, strategy_name):
+    def __init__(self, config, tentacles_setup_config, strategy_name):
         self.is_properly_initialized = False
         self.logger = get_logger(self.get_name())
         self.config = config
+        self.tentacles_setup_config = deepcopy(tentacles_setup_config)
         try:
             from octobot_evaluators.api.evaluators import create_evaluator_classes
             from octobot_evaluators.evaluator.strategy_evaluator import StrategyEvaluator
             from octobot_trading.api.modes import init_trading_mode_config, get_activated_trading_mode
             from tentacles.Evaluator import Strategies
-            init_trading_mode_config(self.config, CONFIG_TRADING_FILE_PATH)
+            init_trading_mode_config(self.config)
             create_evaluator_classes(self.config)
-            self.trading_mode = get_activated_trading_mode(self.config)
+            self.trading_mode = get_activated_trading_mode(self.config, tentacles_setup_config)
```

Comment:
Do we still need `self.config` in `get_activated_trading_mode` with the new `tentacles_setup_config` param ?

Reply:
Yes, we need it to access advanced classes

Reply:
Are they still stored in `config` ?

Reply:
We didn't change that so yep, they are set there by the advanced manager

Reply:
OK do you think it's something that should be changed with the new tentacle manager ?

Reply:
For now the tentacle manager is only about tentacles maintenance and configuration I'm not sure if it makes sense to add advanced classes management to it. I think we should wait a bit to see if we even need these advanced classes as they are currently handled.